warehouse/accounts/views.py

```diff
@@ -118,7 +86,9 @@ def login(request, redirect_field_name=REDIRECT_FIELD_NAME,
 
         # Now that we're logged in we'll want to redirect the user to either
         # where they were trying to go originally, or to the default view.
-        return HTTPSeeOther(redirect_to, headers=dict(headers))
+        return HTTPSeeOther(
+            redirect_to,
+            headers=dict(_login_user(request, userid)))
```

Comment:
Stylistically, I think it would be nicer to split this out so that you have a ``headers = _login_user(request, userid)`` line before this redirect. If only to make the actual step where a user is logged in more obvious and look less like a side effect of generating the redirect.